app/assets/javascripts/discourse/app/routes/topic.js

```diff
@@ -147,13 +149,14 @@ const TopicRoute = DiscourseRoute.extend({
 
   @action
   showHistory(model, revision) {
-    let historyController = showModal("history", {
-      model,
-      modalClass: "history-modal",
+    this.modal.show(HistoryModal, {
+      model: {
+        postId: model.id,
+        postVersion: revision || "latest",
+        post: model,
+        topicController: this.controllerFor("topic"),
```

Comment:
(I realise this was not introduced by your change, but it would be nice to improve while we're here)

Passing the whole topicController in here is not a very clean API. It looks like we're only using the `editPost()` action, so could we do something like this instead?

```suggestion
        editPost: (post) => this.controllerFor("topic").send("editPost", post),
```

That way, the modal itself is isolated from implementation details of the topic controller.